internal/network/connectivity_groups_test.go

```diff
@@ -2,41 +2,76 @@ package network
 
 import (
 	"encoding/json"
+	"fmt"
 
 	"github.com/go-openapi/strfmt"
+	"github.com/google/uuid"
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 	"github.com/openshift/assisted-service/models"
 )
 
-var _ = Describe("connectivity groups", func() {
+type node struct {
+	id          *strfmt.UUID
+	addressNet1 string
+	addressNet2 string
+}
 
-	var (
-		hid1 strfmt.UUID = "11111111-1111-1111-1111-111111111111"
-		hid2 strfmt.UUID = "22222222-2222-2222-2222-222222222222"
-		hid3 strfmt.UUID = "33333333-3333-3333-3333-333333333333"
-		hid4 strfmt.UUID = "44444444-4444-4444-4444-444444444444"
-		hid5 strfmt.UUID = "55555555-5555-5555-5555-555555555555"
-		hid6 strfmt.UUID = "66666666-6666-6666-6666-666666666666"
-		hid7 strfmt.UUID = "77777777-7777-7777-7777-777777777777"
-	)
+func linkNet1(n *node) *models.L2Connectivity {
+	return &models.L2Connectivity{
+		RemoteIPAddress: n.addressNet1,
+		Successful:      true,
+	}
+}
 
-	createL2 := func(outgoingIpAddress string, successful bool) *models.L2Connectivity {
-		return &models.L2Connectivity{
-			OutgoingIPAddress: outgoingIpAddress,
-			Successful:        successful,
-		}
+func linkNet2(n *node) *models.L2Connectivity {
+	return &models.L2Connectivity{
+		RemoteIPAddress: n.addressNet2,
+		Successful:      true,
+	}
+}
+
+func unLinkNet1(n *node) *models.L2Connectivity {
+	return &models.L2Connectivity{
+		RemoteIPAddress: n.addressNet1,
+		Successful:      false,
 	}
+}
+
+func generateNodes(count int) []*node {
 
-	createRemoteHost := func(id strfmt.UUID, l2s ...*models.L2Connectivity) *models.ConnectivityRemoteHost {
-		return &models.ConnectivityRemoteHost{
-			HostID:         id,
-			L2Connectivity: l2s,
+	ret := make([]*node, count+1) // leave [0] empty for easy referencing
+	for i := 0; i < count; i++ {
+		id := strfmt.UUID(uuid.New().String())
+		ret[i+1] = &node{
```

Comment:
Why i + 1?

Reply:
Line 44 "// leave [0] empty for easy referencing"

Reply:
It had hid1,hid2, etc. I kept the convention instead of starting with 0